prototype2/multiblade/mbcommon/MBConfig.h

```diff
@@ -0,0 +1,51 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief class for configuration of multi blade detector settings
+/// reading from json file.
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <mbcaen/MB16Detector.h>
+#include <string>
+#include <vector>
+
+class MBConfig {
+public:
+  enum class InstrumentGeometry {Estia, Freia};
+
+  ///
+  MBConfig(){};
+
+  /// \brief get configuration from file
+  explicit MBConfig(std::string jsonfile);
+
+  bool isConfigLoaded() {return IsConfigLoaded;}
+
+  /// Name of the json configuratio file to load
```

Comment:
While it maybe a bit more work, I would suggest you properly encapsulate and make these variables accessible only through const getters. Just to be safe that you don't accidentally modify the configuration, the compiler is there to help...

Reply:
Grrr.....

Reply:
Yeah, I know... but in general it pays off, kind of like unit tests ;)

Reply:
I'm not convinced in this case, but I'll humour you this time.